### PR TITLE
parallel scraper

### PR DESCRIPTION
**Pull request by f1yegor**
it occurs that ask mbean server is slow. do it in parallel. 
is there reasons why project compiles to target 1.5?

### CODE REVIEW

File: collector/pom.xml [hunk 1]
```diff
@@ -42,8 +42,8 @@
         <artifactId>maven-compiler-plugin</artifactId>
         <version>2.3.2</version>
         <configuration>
-          <source>1.5</source>
-          <target>1.5</target>
+          <source>1.8</source>
```
Comment (brian-brazil) [inline collector/pom.xml]: 2/3 of our users are still on 1.6, 1.8 is barely in use. This needs to stay as-is for many years to come.

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java [hunk 2]
```diff
@@ -2,11 +2,7 @@
 
 import java.io.IOException;
 import java.lang.management.ManagementFactory;
-import java.util.LinkedHashMap;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Set;
-import java.util.TreeSet;
+import java.util.*;
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: Please leave imports explicit.

File: collector/src/main/java/io/prometheus/jmx/JmxScraper.java [hunk 3]
```diff
@@ -73,9 +68,8 @@ public void doScrape() throws Exception {
                 mBeanNames.removeAll(beanConn.queryNames(name, null));
             }
 
-            for (ObjectName name : mBeanNames) {
-                scrapeBean(beanConn, name);
-            }
+            mBeanNames.parallelStream().forEach(name -> scrapeBean(beanConn, name));
```
Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: This is a Java 1.8 construct, please stick strictly to 1.6 features.

Comment (brian-brazil) [inline collector/src/main/java/io/prometheus/jmx/JmxScraper.java]: parallelStream was only added in 1.8, it'll be at least 2020 before 1.8 is so widespread that we can safely assume that everyone uses at least 1.8.

In addition, what's the cpu impact of doing this in parallel? Spiky cpu usage due to monitoring is undesirable.

**Comment by f1yegor** [comment]
Yeah, I afraid because of 1.5. So I'll keep this only in my fork. Feel free to decline because of it.
And yes, it's more spiky.

**Comment by brian-brazil** [comment]
Spiky is bad when it comes to non-critical cpu usage, you almost never want to delay a user request for a monitoring request. If you're finding scrapes are too slow then I recommend you either blacklist mbeans, or increase your scrape interval.

**Pull request closed without merging**